worlds/stardew_valley/regions.py

```diff
@@ -353,8 +354,81 @@ def inverted(self):
+                    if mod_data.flags[entrance] == "BUILDINGS":
+                        loaded_mod_data.append(ConnectionData(entrance, mod_data.connections[entrance],
+                                                              flag=RandomizationFlag.BUILDINGS))
+                    if mod_data.flags[entrance] == "EVERYTHING":
+                        loaded_mod_data.append(ConnectionData(entrance, mod_data.connections[entrance],
+                                                              flag=RandomizationFlag.EVERYTHING))
+            if (mod_data.mod_name == mod) & (data_type == "Region Remover"):
+                for name in mod_data.region_remover:
+                    loaded_mod_data.append(RegionRemoverData(name, mod_data.region_remover[name]))
+            if (mod_data.mod_name == mod) & (data_type == "Connection Remover"):
+                for name in mod_data.connection_remover:
+                    loaded_mod_data.append(ConnectionRemoverData(name, mod_data.region_remover[name]))
+    return loaded_mod_data
+
+
+def region_extend(mod: str, existing_regions: List[Region], new_regions: List[Region], world_options: StardewOptions):
+    if mod not in world_options[options.Mods]:
+        return
+    to_remove = loaded_mod_region_data(mod, "Region Remover", world_options)
+    for regions_to_remove in to_remove:
+        for region in existing_regions:
+            if regions_to_remove.name == region.name:
+                for exits_to_remove in regions_to_remove.exits:
+                    region.exits.remove(exits_to_remove)
+    for new_region in new_regions:
+        existing_region = next(
+            (region for region in existing_regions if region.name == new_region.name), None)
+        if existing_region:
+            if all(exits in existing_region.exits for exits in new_region.exits):
+                continue
+            existing_region.exits.extend(new_region.exits)
+            continue
+
+        existing_regions.append(new_region)
+
+
+def connection_patcher(mod: str, old_connections: List[ConnectionData], world_options: StardewOptions):
+    if mod in world_options[options.Mods]:
+        to_remove = loaded_mod_region_data(mod, "Connection Remover", world_options)
+        for connection in old_connections:
+            for connections_to_remove in to_remove:
+                if connections_to_remove.name == connection.name:
+                    old_connections.remove(connection)
+        old_connections += loaded_mod_region_data(mod, "Connections", world_options)
+
+
 def create_regions(region_factory: RegionFactory, random: Random, world_options: StardewOptions) -> Tuple[
     Iterable[Region], Dict[str, str]]:
+    for mods in world_options[options.Mods]:
+        if not loaded_mod_region_data(mods, "Regions", world_options):
+            continue
+        loaded_region = loaded_mod_region_data(mods, "Regions", world_options)
+        region_extend(mods, stardew_valley_regions, loaded_region, world_options)
```

Comment:
You are passing the original stardew_valley_regions list as a reference everywhere, then editing it on the fly to add stuff to it. But it is a global value that is re-used everywhere. I believe this is the main cause of that stuff you were saying about duplicated regions everywhere. I would go as far as to say that restructuring this in a more organised way, and not editing the globals, would probably make your region remover and connection remover concepts completely obsolete.

We should never need to add and then remove, we should just leave it as it is and work on copies locally.

If you want to, I can re-write some of this to do what I'm talking about. It would probably be easier, but also yesterday you said you wanted to try to get things working on your own, so I wrote this as a comment and you can give it a shot. But if you get stuck or confused, I'm willing to help by explaining more or writing some of it. Don't hesitate to ask